src/lib.rs

```diff
@@ -855,6 +867,59 @@ impl<I, F> StreamingIterator for TakeWhile<I, F>
     }
 }
 
+/// A streaming iterator which will continue with another operator when the first one finishes
+pub struct AndThen<I, J> {
+    it: I,
+    then: J,
+    done: bool,
+}
+
+impl<I, J> StreamingIterator for AndThen<I, J>
+    where I: StreamingIterator,
+          J: StreamingIterator<Item=I::Item>
+{
+    type Item = I::Item;
+
+    #[inline]
+    fn advance(&mut self) {
+        if !self.done {
+            self.it.advance();
```

Comment:
The behavior of calling `advance` after the end of the first iterator is unspecified. You can either fuse `it` or do something like what the standard library does: https://doc.rust-lang.org/src/core/iter/mod.rs.html#521-525